src/backend/distributed/operations/repair_shards.c

```diff
@@ -849,6 +849,14 @@ CopyShardTables(List *shardIntervalList, char *sourceNodeName, int32 sourceNodeP
 
 		List *ddlCommandList = CopyShardCommandList(shardInterval, sourceNodeName,
 													sourceNodePort, includeDataCopy);
+
+		if (PartitionTable(shardInterval->relationId))
```

Comment:
I think this is a tricky change, and we should be mindful about various edge cases. Citus _implicitly_ relies on the following behavior, are we sure that we are not breaking this?

 
As far as I can see, if a parent has an index, and the partition doesn't have one, it inherits from the parent after ATTACH:

```
 

 CREATE TABLE sensors(
 measureid         integer,
 eventdatetime     date,
 measure_data      jsonb) PARTITION BY RANGE(eventdatetime);
-- an index on the parent
CREATE INDEX si0 ON sensors(measureid);
CREATE TABLE sensors_2020_01_02  (measureid         integer,
eventdatetime     date,
measure_data      jsonb);
ALTER TABLE sensors ATTACH PARTITION sensors_2020_01_02  FOR VALUES FROM ('2020-02-01') TO ('2020-03-01');
select indexrelid::regclass from pg_index where indrelid = 'sensors_2020_01_02'::regclass;
            indexrelid            
----------------------------------
 sensors_2020_01_02_measureid_idx
(1 row)
Time: 0.436 ms
```
 

However, if the child table already has an index with the same columns, the index is not inherited from the parent after ATTACH:
 
```
 CREATE TABLE sensors(
 measureid         integer,
 eventdatetime     date,
 measure_data      jsonb) PARTITION BY RANGE(eventdatetime);
-- parent has the same index
CREATE INDEX si0 ON sensors(measureid);
CREATE TABLE sensors_2020_01_02  (measureid         integer,
eventdatetime     date,
measure_data      jsonb);
-- this time partition also has an index
CREATE INDEX si3 ON sensors_2020_01_02(measureid);
ALTER TABLE sensors ATTACH PARTITION sensors_2020_01_02  FOR VALUES FROM ('2020-02-01') TO ('2020-03-01');
-- this time the index from parent IS NOT inherited
select indexrelid::regclass from pg_index where indrelid = 'sensors_2020_01_02'::regclass;
 indexrelid 
------------
 si3
```

Note: I'm not saying that what Citus relies on makes sense, what I say is that let's be careful with all the indexes/constraints etc during the rebalancing, and make sure that this time we have a good plan that we have a clear view / explicit knowledge :)